docs/source/x_ext.rst

```diff
@@ -557,32 +560,42 @@ Commit interface
   |                    |                        | that the offloaded instruction (``id``) is no longer speculative, will not get killed (e.g. due to misspeculation or an      |
   |                    |                        | exception in a preceding instruction), and is allowed to be committed. If ``commit_valid`` is 1 and ``commit_kill`` is       |
   |                    |                        | 1, then the offloaded instruction (``id``) shall be killed in the |coprocessor| and the |coprocessor| must guarantee that the|
-  |                    |                        | related instruction does/did not change architectural state.                                                                 |
+  |                    |                        | Shall an offloaded instruction be commited or killed?                                                                        |
+  |                    |                        | If ``commit_valid`` is 1 and ``commit_kill`` is 0,  then the core guarantees that the offloaded instruction (``id``) and any |
+  |                    |                        | older (i.e. preceding) instructions are no longer speculative, will not get killed (e.g. due to misspeculation or an        |
+  |                    |                        | exception in a preceding instruction), and are allowed to be committed.                                                      |
+  |                    |                        | If ``commit_valid`` is 1 and ``commit_kill`` is 1, then the offloaded instruction (``id``) and any newer (i.e. succeeding)   |
+  |                    |                        | instructions shall be killed in the |coprocessor| and the |coprocessor| must guarantee that the related instructions do/did  |
+  |                    |                        | not change architectural state.                                                                                              |
   +--------------------+------------------------+------------------------------------------------------------------------------------------------------------------------------+
 
-The ``commit_valid`` signal will be 1 exactly one ``clk`` cycle for every offloaded instruction by the |coprocessor| (whether accepted or not). The ``id`` value indicates which offloaded
-instruction is allowed to be committed or is supposed to be killed.
+The ``commit_valid`` signal will be 1 exactly one ``clk`` cycle.
+It is not required that a commit transaction is performed for each offloaded instruction individually.
+Instructions can be signalled to be non-speculative or to be killed in batch.
+E.g. signalling the oldest instruction to be killed is equivalent to requesting a flush of the |coprocessor|.
+The first instruction to be considered not-to-be-killed after a commit transaction with ``commit_kill`` as 1,
```

Comment:
I still don't think this is correct. I would think that normally the instruction currently getting issued should get killed as well. In this definition it is partially up to the coprocessor whether this instruction will get killed or not (as the coprocessor can decide to accept the instruction the cycle after a kill) and we cannot leave this to the coprocessor.

You could say that the processor could retract the issue request if it wants the currently being issued instruction to be killed. That would however still not solve the scenario in which a processor does not want this instruction being killed (as again the coprocessor would be in control).

Reply:
the typical scenario for a killing of instructions (mispredicted branch, exception), would always lead to the instruction currently hanging in the issue interface to be invalid. I think this is in line with what you are stating. Question is just how to achieve this.

My assumption is that ``issue_valid`` is deasserted (i.e. aborting the transaction) at the same time as ``commit_valid`` is asserted. That leaves nothing to the coprocessor. If the processor doesn't abort by deasserting ``issue_valid``, I assume it is still interested in offloading that.

Practically speaking, are there scenarios in which the CPU would not deassert valid? We might even require that. 

However, I am one the same page as you. We can also reword to
> [...] with successful issue transaction _starting_ at least one clock cycle later

Reply:
solved in 8081293